File: 830-largest-triangle-area/largest-triangle-area.py
class Solution:
    def largestTriangleArea(self, points: List[List[int]]) -> float:
        n = len(points)
        if n < 3: return 0
        maxArea = 0
        for i in range(n):
            for j in range(n):
                for k in range(n):
                    a = abs(((points[i][0]-points[j][0])**2 +(points[i][1]-points[j][1])**2)**0.5)
                    b = abs(((points[k][0]-points[j][0])**2 +(points[k][1]-points[j][1])**2)**0.5)
                    c = abs(((points[i][0]-points[k][0])**2 +(points[i][1]-points[k][1])**2)**0.5)
                    s = (a+b+c)/2
                    if s<a or s<b or s<c: continue
                    area = (s*(s-a)*(s-b)*(s-c))**0.5
                    maxArea = max(area,maxArea)
        return maxArea

File: 830-largest-triangle-area/largest-triangle-area.cpp
class Solution {
public:
    double largestTriangleArea(vector<vector<int>>& points) {
        double area =0;
        int n = points.size();
        if (n < 3) return area;
        for (int i=0; i<n; i++){
            int x1 = points[i][0] ,y1 = points[i][1];
            for (int j=i+1; j<n; j++){
                int x2 = points[j][0], y2 = points[j][1];
                for(int k=j+1; k<n; k++){
                    int x3 = points[k][0], y3 = points[k][1];
                    area = std::max(area, 0.5*(abs(x1*(y2-y3) + x2*(y3-y1) + x3*(y1-y2))));
                }
            }
        }
        return area;
    }
};